autowarefoundation/autoware.universe
# feat(ad_service_state_monitor)!: remove ad_service_state_monitor

## Description

Remove ad_service_state_monitor and add the autoware state node for compatibility. The functionality of ad_service_state_monitor has been reorganized into the autoware state node and the following packages.
- state machine: `/autoware/state` is now converted in autoware state node from `initialization state`, `routing state` and `operation mode` provided by AD API
    - finalizing state: `/autoware/shutdown` service will continue to be supported in autoware state node.
    - arrived goal: The goal arrival check is moved to `mission_planner` package.
    - topic check: The topic rate check is moved to `component_state_monitor` package.
- clear route: `/autoware/reset_route`is removed. Use `/api/routing/clear_route` instead.
- disengage_on_goal: The routing node in default_ad_api changes operation mode to stop when the vehicle goals.
- disengage_on_route:  The routing node in default_ad_api changes operation mode to stop when `/api/routing/clear_route` .

## Related links

https://github.com/tier4/autoware_launch/pull/568

## Tests performed

Using simple_planning_simulator.
- Check if the vehicle can drive and if the state transitions as before.
- Check if the operation mode will be `stop` when the vehicle goals.
- Check if the operation mode will be `stop` when the route is cleared (while the vehicle is driving).
- Check if the autoware state will be `finalizing` when `/autoware/shutdown` service is called.

## Notes for reviewers

None

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.


## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.


## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.


After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/
